yarpl/include/yarpl/single/SingleOperator.h

```diff
@@ -97,33 +96,33 @@ template <
     typename D,
     typename F,
     typename = typename std::enable_if<std::is_callable<F(U), D>::value>::type>
-class MapOperator : public SingleOperator<U, D> {
+class MapOperator : public SingleOperator<U, D, MapOperator<U, D, F>> {
+  using ThisMap = MapOperator<U, D, F>;
```

Comment:
please consider renaming ThisMap to ThisOperatorT
MSuper to Super
and removing SSuper or renaming SSuper to ThisOperatorSubscriptionT

just improving readability...